perun-core/src/main/java/cz/metacentrum/perun/core/api/AuthzResolver.java

```diff
@@ -590,6 +592,22 @@ public static boolean isAuthorizedToManageRole(PerunSession sess, PerunBean comp
 		return hasOneOfTheRolesForObject(sess, complementaryObject, Privileges.getRolesWhichCanManageRole(role));
 	}
 
+	/**
+	 * Check wheter the principal is authorized to manage the role on the object.
+	 *
+	 * @param sess principal's perun session
+	 * @param complementaryObject bounded with the role
+	 * @param role which will be managed
+	 * @return
+	 * @throws RoleManagementRulesNotExistsException when the role does not have the management rules.
+	 */
+	public static boolean authorizedToManageRole(PerunSession sess, PerunBean complementaryObject, String role) throws RoleManagementRulesNotExistsException {
```

Comment:
We have discussed this and the result is that we do not want to support such roles. It could result in undesired authorization decisions and it would make the whole authorization even more complex.
The solution for this role would be to create VoCommunicator (Vo) and FacilityCommunicator (Facility). You can create AND rule for them if you need that.